src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.cs

```diff
@@ -864,6 +864,23 @@ private static bool PackBlocks(ArrayBuilder<BasicBlockBuilder> blocks, PooledDic
                                         continue;
                                     }
 
+                                    if (currentRegion.LastBlock == block &&
```

Comment:
@AlekseyTs rather than adjust the verifier to allow this case, I opted to not pack the block instead. This affects `SwitchExpression_JustDiscardWithGuard` B2 and B4: without this condition, they would be packed into just B2. I can take the approach of the verifier instead, but this seemed the simpler and slightly-more-correct solution.